query-engine/connector-test-kit-rs/query-engine-tests/tests/raw/sql/input_coercion.rs

```diff
@@ -0,0 +1,64 @@
+use query_engine_tests::*;
+
+#[test_suite(schema(common_nullable_types), only(Postgres))]
+mod input_coercion {
+    use query_engine_tests::fmt_execute_raw;
+
+    // Checks that query raw inputs are coerced to the correct types
+    #[connector_test]
+    async fn scalar_input_correctly_coerced(runner: Runner) -> TestResult<()> {
+        run_query!(
+            &runner,
+            fmt_execute_raw(
+                r#"INSERT INTO "TestModel" ("id", "string", "int", "bInt", "float", "bytes", "bool", "dt") VALUES ($1, $2, $3, $4, $5, $6, $7, $8);"#,
+                vec![
+                    RawParam::scalar(1_i32),
+                    RawParam::scalar("str"),
+                    RawParam::scalar(42_i32),
+                    RawParam::bigint(9223372036854775807),
+                    RawParam::scalar(1.5432_f64),
+                    RawParam::bytes(&[1, 2, 3]),
```

Comment:
Not really a big problem, but bigint etc are also scalars, the naming is off.

Reply:
I'll just use `RawParam::from` for all primitive types